addons/website_sale_delivery/models/sale_order.py

```diff
@@ -72,3 +71,46 @@ def _cart_update(self, *args, **kwargs):
         """ Override to update carrier quotation if quantity changed """
         self._remove_delivery_line()
         return super()._cart_update(*args, **kwargs)
+
+    def _action_confirm(self):
+        for order in self:
+            order_location = order.access_point_address
+
+            if not order_location:
+                continue
+
+            # retreive all the data :
+            # name, street, city, state, zip, country
+            name = order.partner_shipping_id.name
+            street = order_location['pick_up_point_address']
+            city = order_location['pick_up_point_town']
+            zip_code = order_location['pick_up_point_postal_code']
+            country = order.env['res.country'].search([('code', '=', order_location['pick_up_point_country'])]).id
+            state = order.env['res.country.state'].search(['&', ('code', '=', order_location['pick_up_point_state']), ('country_id.id', '=', country)]).id if (order_location['pick_up_point_state'] and country)else None
+            parent_id = order.partner_shipping_id.id
+            email = order.partner_shipping_id.email
+            phone = order.partner_shipping_id.phone
+
+            # we can check if the current partner has a partner of type "delivery" that has the same address
+            existing_partner = order.env['res.partner'].search(['&', '&', '&', ('street', '=', street), ('city', '=', city), ('state_id', '=', state), ('country_id', '=', country)])
+
+            if existing_partner:
+                if len(existing_partner) > 1:
+                    order.partner_shipping_id = existing_partner[0]
+                else:
+                    order.partner_shipping_id = existing_partner
```

Comment:
This is a bit redundant
```suggestion
                order.partner_shipping_id = existing_partner[0]
```